compiler/rustc_codegen_ssa/src/traits/consts.rs

```diff
@@ -26,6 +26,8 @@ pub trait ConstMethods<'tcx>: BackendTypes {
     fn const_to_opt_uint(&self, v: Self::Value) -> Option<u64>;
     fn const_to_opt_u128(&self, v: Self::Value, sign_ext: bool) -> Option<u128>;
 
+    fn const_data_from_alloc(&self, alloc: &Allocation) -> Self::Value;
```

Comment:
This is just `from_const_alloc`, isn't it? Just with a hardcoded `offset` of `0` and no type.

Reply:
`from_const_alloc` also returns a `PlaceRef` while this returns a raw pointer value.

Reply:
Right, but the `PlaceRef` is made from that pointer value and the type/layout passed in.